src/Route.php

```diff
@@ -170,10 +174,31 @@ public function defaults(array $defaults): self
 
     /**
      * @param callable|MiddlewareInterface $middleware
-     * @return MiddlewareInterface
+     * @return MiddlewareInterface|string|array
      */
-    private function prepareMiddleware($middleware): MiddlewareInterface
+    private function prepareMiddleware($middleware)
     {
+        if (
+            is_string($middleware) && class_exists($middleware)
+            && in_array(MiddlewareInterface::class, class_implements($middleware))
+        ) {
+            if ($this->container === null) {
+                throw new InvalidArgumentException('Route container must not be null for lazy loaded middleware.');
+            }
+            return $middleware;
+        }
+
+        if (
+            is_array($middleware) && count($middleware) === 2
+            && isset($middleware[0]) && isset($middleware[1])
+            && is_string($middleware[1]) && is_string($middleware[0]) && class_exists($middleware[0])
```

Comment:
Do we need to check if a class exists?

Reply:
Yes, I think that in this way we can get a more informative error message for the developer.